web-src/src/components/FavoriteVariantCard.js

```diff
@@ -37,21 +46,58 @@ const styles = {
 };
 
 export function FavoriteVariantCard({ variant, ...props }) {
+  const { firefallService, expressSDKService } = useApplicationContext();
+  const { isExpressAuthorized } = useShellContext();
   const toggleFavorite = useToggleFavorite();
+  const { addImageToVariant } = useVariantImages();
+
+  const [imagePromptProgress, setImagePromptProgress] = useState(false);
+
+  const handleGenerateImage = useCallback(async (imagePrompt) => {
```

Comment:
is there a value in extracting this logic into a single place ? I noticed it's used twice, but I also personally prefer clarity in the code at the cost of a little duplication, unless there's a value to manage the logic in a single place (making changes in behavior easier)

Reply:
I actually thought about extracting the logic into a single place but didn't value in this because the function has a lot of dependency within the component.